web/src/client/network/network_manager.js

```diff
@@ -500,19 +502,29 @@ class NetworkManagerAdapter {
     return { address: data.address.v, prefix: parseInt(data.prefix.v) };
   }
 
-  settingsFromProxies(manager, settings) {
-    return {
-      wireless: !!(manager?.WirelessEnabled && manager?.WirelessHardwareEnabled),
-      hostname: settings?.Hostname || ""
-    };
+  /*
+  * Returns whether the system is able to scan wifi networks based on rfkill and the presence of
+  * some wifi device
+  *
+  * @return {boolean}
+  */
+  wirelessScanSupported() {
+    const wEnabled = !!(this.proxies.manager?.WirelessEnabled && this.proxies.manager?.WirelessHardwareEnabled);
```

Comment:
It looks a little hard to read to me. I would rather prefer something like (not tested):

```javascript
if (this.proxies.manager === undefined) return false;

const { WirelessEnabled: wirelessEnabled, WirelessHardwareEnabled: wirelessHardwareEnabled} = this.proxies.manager;
const wifiDevices = Object.values(this.proxies.devices).filter(d => d.DeviceType === NM_DEVICE_TYPE_WIFI);
return WirelessEnabled && wirelessHardware && wifiDevices.length > 0;
```

So return earlier if the manager is unavailable, and make the conditions more readable. Additionally, why do you use "wireless" and "wifi"? It does not look consistent.

Reply:
Originally it represented the state of rfill enablement but that is not the case anymore, so, the method and variable should be renamed to WiFi as we are not interested in other kind of devices by now.

Reply:
Adapted, I basically renamed the settings property and the method to hopefully a more meaningful name.